### ISSUE TITLE
Unable to get temp file on Windows

### ISSUE THREAD

**Issue by kerbo** [jruby/jruby#3430]
This issue was introduced in 9.0.3.0. When trying to upload a file, I get this error from Rack Multipart upload:

Unable to find a non world-writable directory for Dir::tmpdir. Consider setting ENV['TMPDIR'], ENV['TMP'] or ENV['TEMP'] to a non world-writable directory.

I reverted back to 9.0.1.0 and everything is working again.

**Comment by hathawad** [comment]
I get this when trying to run rails console, so can't even initialize Rails on 9.0.3.0

**Environment:**
```log
jruby 9.0.3.0 (2.2.2) 2015-10-21 633c9aa Java HotSpot(TM) 64-Bit Server VM 24.80-b11 on 1.7.0_80-b15 +jit [Windows Server 2008 R2-amd64]
```

**Error:**

```log
ArgumentError: Unable to find a non world-writable directory for Dir::tmpdir. Co
nsider setting ENV['TMPDIR'], ENV['TMP'] or ENV['TEMP'] to a non world-writable
directory.
                     tmpdir at C:/jruby-9.0.3.0/lib/ruby/stdlib/tmpdir.rb:41
                 initialize at ..../lib/rack/raw_upload.rb:12

**Comment by headius** [comment]
Hmmm I wonder if this is the fstat bug we introduced, @enebo.

**Comment by djberg96** [comment]
@headius Is the fstat bug what's causing this when I try to run jirb with 9.0.3.0?

```
Unhandled Java exception: java.lang.IllegalArgumentException: instance of class jnr.posix.WindowsRawFileStat is not a struct
java.lang.IllegalArgumentException: instance of class jnr.posix.WindowsRawFileStat is not a struct
toNative at jnr/posix/BaseNativePOSIX.java:709
toNative at jnr/posix/BaseNativePOSIX.java:696
   fstat at jnr/posix/BaseNativePOSIX.java:135
   fstat at jnr/posix/BaseNativePOSIX.java:124
   fstat at jnr/posix/CheckedPOSIX.java:125
   fstat at jnr/posix/LazyPOSIX.java:122
```

**Comment by kerbo** [comment]
This issue is fixed in 9.0.4.0
